dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/ServiceCaller.java

```diff
@@ -462,6 +476,17 @@ private Optional<ReferenceAndService> trackCurrent() {
 
     private Optional<ReferenceAndService> getCurrent() {
         BundleContext context = getContext();
+        if (System.getSecurityManager() != null && bundleHasPermission(context, servicePermission)) {
+            return AccessController.doPrivileged(
+                                                 (PrivilegedAction<Optional<ReferenceAndService>>) () -> {
+                                                     return getCurrentInternal(context);
+                                                 });
+        } else {
+            return getCurrentInternal(context);
```

Comment:
I think this is fine because it should end up throwing a `SecurityException` if the security manager is not `null` and the calling bundle did not have permissions above.  Maybe add a comment if that was the intention here.

Reply:
That was the intention.